### ISSUE TITLE
Message order is not preserved in cluster

### ISSUE THREAD

**Issue by arnecs-piscada** [emqx/emqx#3046]
#### Environment

- OS: Docker emqx/emqx:v3.2.5
- Erlang/OTP: 
- EMQ: 3.2.5

#### Description
Message order is not preserved between brokers in cluster, when consecutively published to the same topic.
When two clients are connected to the same broker, all messages are received in the order they are sent. However when two clients are connected to two different brokers, not all messages are receieved in the same order they are sent.

It seems like the order of messages is not preserved when routing the messages between brokers.

**Comment by tigercl** [comment]
@arnecs-piscada 
EMQ X guarantees the order of messages even in cluster. May you provide more details about how you send and receive?

**Comment by velimir** [comment]
Related to #3000 ?

**Comment by arnecs-piscada** [comment]
By using `mosquitto_pub` and `mosquitto_sub`. 
We have a cluster of 3 brokers running in kubernetes.

Subscribe to broker `emqx-0`
```bash
mosquitto_sub -h emqx-0 -p 1883 -t test
```

Publish messages with a incrementing number in payload to broker `emqx-1`.
```bash
printf "%d\n" {1..5} | mosquitto_pub -h emqx-1 -p 1883 -t test -q 1 -l
```
**Expected output**
```
1
2
3
4
5
```

**Actual** 
```
1
2
3
5
4
```

Output from the subscriber shows that the messages is not always received in the same order they got sent. Mulitple client connected to the same broker always receive the messages in the same order. Having a client connected to the third broker may receive the messages in a different order.

**Comment by tigercl** [comment]
@arnecs-piscada You are right, we only fixed it in v4.0 due to our negligence. We will synchronize the repair to v3.2.6 as soon as possible, thanks for your feedback.

**Comment by Utkagr** [comment]
> @arnecs-piscada You are right, we only fixed it in v4.0 due to our negligence. We will synchronize the repair to v3.2.6 as soon as possible, thanks for your feedback.

Hi @tigercl, how do we know when the repair is synchronized to v3.2.6? Is it done already? Thanks.
